wordpress-mobile/WordPress-iOS
# Issue/5077 improve media error messages

Fixes #5077

This PR improves error handling of media uploads that fail because of Quota limitations for sites where those are unknown or not available ( self-hosted sites).

During the development of this I also found some race conditions between the reporting of a media failure and progress reporting that where causing incorrect states on the Media Library. This PR also does some changes to address those issues by making sure the `remoteStatus` of media is updated before failure is reported back,

To test:
 - On a self-hosted site with a low disk quota and or small max file upload size
 - Try to upload a large media that goes over the limits
 - See if good human readable error messages show reporting the reason of the upload failing.
